### ISSUE TITLE
Integrate REUSE standard

### ISSUE THREAD

**Issue by NicolasToussaint** [fossology/fossology#1592]
### Description

REUSE (https://reuse.software/) proposes a way to standardize the way licence and copyrights informations are stored within a project source code.

For example, it expects
- all text files to include SPDX copyright & licences identifiers, 
- binary files to be accompanied by a `foo.jpg.license` file
- all licence texts to be available under a folder `LICENSES`

The projects provides a simple python tool that validates that the project is REUSE compliant, and points the files for which some information is missing if not.

#### Steps followed and expected result

Fossology could:
1. (easy) perform this check and
   - indicate  if an upload is REUSE compliant or not
   - point to the files that need fixing
2. (not so easy) Given the licensing and copyright information gathered thanks to the clearing process, add the required files to the project. (maybe propose to download a copy of the source code in which this data has been injected?)

**Comment by silverhook** [comment]
@NicolasToussaint, I would say the new Ojo agent addresses this. Did you give it a try already and if so, please either close this issue or provide further info how to improve Ojo.

**Comment by NicolasToussaint** [comment]
Hi @silverhook,

I guess Ojo provides part of the solution, but I think that some aspects of the REUSE specifications are not addressed.
1. detecting the files like `foo.jpg.license`
1. handling the `dep5` file
1. Verifying that the required license texts are present in the `LICENSES` folder

I suppose that Ojo could be enhanced to fill the missing items.

Maybe @mxmehl can help identify what else is missing to fully support the REUSE specifications ?


An extra option would be to display that a project is fully REUSE compliant (though I have know idea where).

**Comment by silverhook** [comment]
Pulling in also @carmenbianca

**Comment by carmenbianca** [comment]
I'm fairly certain that confirming REUSE compliance is out of the scope of FOSSology. The following things seem in-scope, however:

- Detecting the SPDX tags recommended by REUSE. (SPDX-License-Identifier, SPDX-FileCopyrightText).
- Detecting .license files.
- Detecting the LICENSES/ directory.
- Parsing `.reuse/dep5`, possibly.

**Comment by NicolasToussaint** [comment]
I agree that displaying the REUSE conformance inside Fossology is not the ultimate goal - I just saw that as a cheap and nice bonus.

If the the 3 first points you mention are handled (not sure the  `dep5` is worth the work either), all REUSE conformant uploads will turn green anyway, and that's what we're after.

**Comment by mxmehl** [comment]
For the record, I agree to the list brought forward by @carmenbianca. 

I'm not sure whether the check of REUSE conformance inside FOSSology makes sense as the REUSE helper tool is more or less the safest, most up-to-date technique (with the specification) to confirm this. If we ever changed bits of the specification, we would also have to update FOSSology accordingly.

**Comment by NicolasToussaint** [comment]
Recap on this Issue:

- Detecting the SPDX tags recommended by REUSE.
  - [x] SPDX-License-Identifier -> Done by Ojo agent
  - [ ] SPDX-FileCopyrightText -> To Do
- [x] Detecting .license files -> Done by ReSo agent (#1833)
- [ ] Detecting the LICENSES/ directory -> To Do
- Parsing .reuse/dep5, possibly.

**Comment by silverhook** [comment]
This issue (esp. the `{$file}.license` being ignored) popped up again at the FOSDEM Tooling Fringe event.

**Comment by NicolasToussaint** [comment]
Maybe this one should be closed, as the remaining item is tracked in #2214

**Comment by mxmehl** [comment]
As a less-than-sporadic user of Fossology I have to ask: is the [REUSE.toml](https://reuse.software/spec-3.3/#reusetoml) integrated already?
